src/python/pants/backend/cc/dependency_inference/rules.py

```diff
@@ -175,7 +223,15 @@ async def infer_cc_source_dependencies(
                 maybe_disambiguated = explicitly_provided_deps.disambiguated(ambiguous)
                 if maybe_disambiguated:
                     result.add(maybe_disambiguated)
+                    continue
+
+        # Finally, check the alternative source roots
+        logger.debug(f"Checking alternatives for {include.path}")
+        subdir = cc_files_mapping.subdirectory_mapping.get(include.path)
```

Comment:
Generally in other areas of Pants, multiple sources for a match would be put together first and then checked for ambiguity. For my education, what does it mean for an include to be processed this way?

Reply:
This is just the last case looking for dependencies, which wasn't handled alternatively.

This covers the cases where there is an include directory, and the headers are labeled something like `core/foo.h` - which aren't system headers, not relative headers, but the "other" headers (no idea what the technical term for this is - I would have called them namespaced headers, except `namespace` is its own thing).